controller/users.go

```diff
@@ -756,8 +758,23 @@ func (c *UsersController) UpdateByServiceAccount(ctx *app.UpdateByServiceAccount
 			return tr.Users().Save(ctx, user)
 		})
 		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"err":         err,
+				"identity_id": ctx.ID,
+			}, "unable to deprovision user in Auth DB")
 			return jsonapi.JSONErrorResponse(ctx, errors.NewInternalErrorFromString(ctx, err.Error()))
 		}
+		if c.tenantService != nil {
+			// Delete tenant
+			err = c.tenantService.Delete(ctx, identity.ID)
+			if err != nil {
+				log.Error(ctx, map[string]interface{}{
+					"err":         err,
+					"identity_id": ctx.ID,
+				}, "unable to delete tenant when deprovisioning user")
+				// Log the error and proceed
```

Comment:
Good! The workflow shouldn't hit a wall if for some reason the `delete tenant` fails.